pontis-package/pontis/core/utils.py

```diff
@@ -23,5 +23,14 @@ def currency_pair_to_key(quote, base):
 
 def pprint_entry(entry):
     print(
-        f"Entry: key={felt_to_str(entry.key)}, value={entry.value}, timestamp={entry.timestamp}, publisher={felt_to_str(entry.publisher)}"
+        f"Entry: key={felt_to_str(entry.key)}, value={entry.value}, timestamp={entry.timestamp}, source={entry.source}, publisher={felt_to_str(entry.publisher)}"
     )
+
+
+def key_for_asset(asset):
+    if "key" in asset:
+        key = currency_pair_to_key(*asset["pair"])
+    else:
+        key = asset["key"]
+
+    return key
```

Comment:
Not in this PR, but would be great to introduce stricter typing here going forward

Reply:
Agree